### PR TITLE
HBW-37 Show bp status

### PR DESCRIPTION
**Pull request by GideonEXE**

### CODE REVIEW

File: hbw/app/models/hbw/widget.rb [hunk 1]
```diff
@@ -17,12 +17,16 @@ def entity_type_buttons(entity_type)
 
   include HBW::Inject[:adapter]
 
-  def bp_buttons(entity_identifier, entity_type)
-    return [] if @adapter.bp_running?(entity_identifier)
+  def bp_buttons(entity_identifier, entity_type, current_user_identifier)
+    return {} if !@adapter.user_exist?(current_user_identifier)
```
Comment (sugarfree1) [inline hbw/app/models/hbw/widget.rb]: Why don't you rewrite it as
```
if ...
elsif ...
else
```
?
It's a better practice, than writing 3 ```return```s.

File: hbw/app/models/hbw/task.rb [hunk 2]
```diff
@@ -13,21 +13,24 @@ def method_missing(variable_name)
...
-                     includeProcessVariables: true,
-                     processInstanceVariables: [
-                       name: HBW::Widget.config['entity_code_key'],
-                       operation: operation,
-                       value: entity_code
-                     ],
-                     size: size))
+        assignee = b_for_all_users ? nil : user.try(:id) || email
+        if !user.nil?
+          operation = entity_code == '%' ? :like : :equals
```
Comment (sugarfree1) [inline hbw/app/models/hbw/task.rb]: Move it to separate method.

File: hbw/app/models/hbw/task.rb [hunk 3]
```diff
@@ -13,21 +13,24 @@ def method_missing(variable_name)
...
-                     active: true,
-                     includeProcessVariables: true,
-                     processInstanceVariables: [
-                       name: HBW::Widget.config['entity_code_key'],
-                       operation: operation,
-                       value: entity_code
-                     ],
-                     size: size))
+        assignee = b_for_all_users ? nil : user.try(:id) || email
+        if !user.nil?
```
Comment (sugarfree1) [inline hbw/app/models/hbw/task.rb]: ```unless```

File: hbw/app/models/hbw/task.rb [hunk 4]
```diff
@@ -13,21 +13,24 @@ def method_missing(variable_name)
...
-                     assignee: user.try(:id) || email,
-                     active: true,
-                     includeProcessVariables: true,
-                     processInstanceVariables: [
-                       name: HBW::Widget.config['entity_code_key'],
-                       operation: operation,
-                       value: entity_code
-                     ],
-                     size: size))
+        assignee = b_for_all_users ? nil : user.try(:id) || email
```
Comment (sugarfree1) [inline hbw/app/models/hbw/task.rb]: Move it to separate method.

File: hbw/app/models/hbw/task.rb [hunk 5]
```diff
@@ -13,21 +13,24 @@ def method_missing(variable_name)
 
     class << self
       using_connection \
-      def fetch(email, entity_code, size = 1000)
+      def fetch(email, entity_code, size = 1000, b_for_all_users = false)
```
Comment (sugarfree1) [inline hbw/app/models/hbw/task.rb]: What is ```b_```?

File: hbw/app/assets/javascripts/hbw/components/buttons.js.jsx.coffee [hunk 6]
```diff
@@ -67,20 +68,23 @@ modulejs.define(
         @state.subscription.close()
 
       render: ->
-        if @props.showSpinner or !@state.fetched
-          `<div className="hbw-spinner"><i className="fa fa-spinner fa-spin fa-2x"></i></div>`
+        if @props.env.userExist
+          if @props.showSpinner or !@state.fetched or @state.bpRunning
```
Comment (sugarfree1) [inline hbw/app/assets/javascripts/hbw/components/buttons.js.jsx.coffee]: When this condition ```!@state.fetched``` is true?

File: hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee [hunk 7]
```diff
@@ -84,5 +85,8 @@ modulejs.define(
           url: @env.connection.serverURL + '/users/check'
           method: 'GET'
           contentType: 'application/json').done((data) =>
-            @env.dispatcher.trigger('hbw:activiti-user-not-found') if !data.user_exist)
+            if !data.user_exist
```
Comment (sugarfree1) [inline hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee]: ```unless```?

**Comment by GideonEXE** [comment]
UPD

File: hbw/app/models/hbw/task.rb [hunk 8]
```diff
@@ -38,6 +39,22 @@ def wrap(tasks)
         tasks.map { |task|
           new(task.merge('processDefinition' => definitions[task.fetch('processDefinitionUrl')])) }
       end
+
+      def assignee(user, email, for_all_users)
+        if for_all_users
```
Comment (sugarfree1) [inline hbw/app/models/hbw/task.rb]:
```
unless for_all_users
  user.try(:id) || email
end
```

**Comment by GideonEXE** [comment]
UPD

**Comment by sugarfree1** [comment]
merge

**Pull request merged**